src/platform/android/DnssdImpl.cpp

```diff
@@ -183,6 +183,25 @@ CHIP_ERROR ChipDnssdStopBrowse(intptr_t browseIdentifier)
     return CHIP_ERROR_NOT_IMPLEMENTED;
 }
 
+template <size_t N>
+CHIP_ERROR extractProtocol(const char * serviceType, char (&outServiceName)[N], DnssdServiceProtocol & outProtocol)
+{
+    std::string serviceTypeStr(serviceType);
+    size_t index = serviceTypeStr.find(".");
+
+    ReturnErrorCodeIf(index == std::string::npos, CHIP_ERROR_INVALID_ARGUMENT);
+    ReturnErrorCodeIf(index + 1 > N, CHIP_ERROR_INVALID_ARGUMENT);
+
+    memcpy(outServiceName, serviceType, index);
```

Comment:
why do we need to return the string actually?

if we return DnssdServiceProtocol it is trivial to have:

```cpp
const char *ProtocolString(DnssdServiceProtocol p) {
  if (p == DnssdServiceProtocol::kDnssdProtocolTcp) {
     return "_tcp";
  }
  VerifyOrDie(p == DnssdServiceProtocol::kDnssdProtocolUdp);
  return "_udp";
}
```

we seem to otherwise return the same thing in two separate ways, making more complex methods.


Also agree with the point below from @tcarmelveilleux : why does code care about protocol? I thought DNSSD strings are just opaque for our purposes for this browsing prupose.

Reply:
1. I modified the code by referring to your additional comment and the Linux platform code below. Please check.
https://github.com/project-chip/connectedhomeip/blob/733afaabd512cbf396d3edf09805d1054a6ec5f8/src/platform/Linux/DnssdImpl.cpp#L569

2. As I explained above, the value is not set only for Android Platform. Therefore, I think there is a problem with not knowing which value will be set.